cli/azd/pkg/account/subscriptions_cache.go

```diff
@@ -0,0 +1,75 @@
+package account
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+)
+
+// The storage cache used for storing subscriptions accessible by the currently logged in user.
+const cSubscriptionsCacheFile = "subscriptions.cache"
+
+// SubscriptionsCache caches subscriptions with the tenant required to access each subscription for the logged in account.
+//
+// The cache is backed by an in-memory copy, then by local file system storage.
+type SubscriptionsCache struct {
+	cachePath string
+
+	inMemoryCopy []Subscription
+}
+
+func NewSubscriptionsCache() (*SubscriptionsCache, error) {
+	configDir, err := config.GetUserConfigDir()
+	if err != nil {
+		return nil, fmt.Errorf("loading stored user subscriptions: %w", err)
+	}
+
+	return NewSubscriptionsCacheWithDir(filepath.Join(configDir, cSubscriptionsCacheFile))
+}
+
+func NewSubscriptionsCacheWithDir(cachePath string) (*SubscriptionsCache, error) {
+	return &SubscriptionsCache{
+		cachePath: cachePath,
+	}, nil
+}
+
+// Load loads the subscriptions from cache.
+func (s *SubscriptionsCache) Load() ([]Subscription, error) {
+	if s.inMemoryCopy != nil {
+		return s.inMemoryCopy, nil
+	}
+
+	cacheFile, err := os.ReadFile(s.cachePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var subscriptions []Subscription
+	err = json.Unmarshal(cacheFile, &subscriptions)
+	if err != nil {
+		return nil, err
+	}
+
+	s.inMemoryCopy = subscriptions
+	return subscriptions, nil
```

Comment:
It's not a good experience, since it's not a config and meant to be removed / recreated. It also can end up expanding the config file by a ton, and none of this is actual configurable configuration.